stress/src/throughput.rs

```diff
@@ -28,7 +29,28 @@ where
         STOP.store(true, Ordering::SeqCst);
     })
     .expect("Error setting Ctrl-C handler");
-    let num_threads = num_cpus::get();
+
+    let mut num_threads = num_cpus::get();
+    let mut args_iter = env::args();
+
+    if let Some(arg_str) = args_iter.nth(1) {
+        let arg = arg_str.parse::<usize>();
+
+        if !arg.is_ok() {
+            eprintln!("Invalid command line argument '{}' as number of threads. Make sure the value is a positive integer.", arg_str);
+            std::process::exit(1);
+        }
+
+        let arg_num = arg.unwrap();
+
+        if arg_num > 0 && arg_num <= num_threads {
```

Comment:
I think when the user has provided a value greater than the number of logical CPUs, we should run the stress test with the maximum allowed value instead of simply exiting. We could log a message saying something like "the value provided was higher than the number of logical cores on the machine. Stress test would be run with n threads instead."

Reply:
Ok. Updated, stay with the passed in threads even if it is larger than the available logical cores.